src/main/scala/net/ruippeixotog/scalascraper/browser/Browser.scala

```diff
@@ -15,6 +15,8 @@ trait Browser {
   def parseFile(path: String, charset: String): Document = parseFile(new File(path), charset)
   def parseFile(path: String): Document = parseFile(new File(path), "UTF-8")
 
+  def parseResource(name: String, charset: String = "UTF-8"): Document = parseFile(new File(getClass.getResource(name).toURI), charset)
```

Comment:
I would actually prefer the handling of the input stream to be delegated to each `Browser` implementation, as both Jsoup and HtmlUnit have their ways to deal with them.

Right now `Browser` has both `parseString` and `parseFile` as abstract methods, even though one could be implemented using the other. I believe a third `parseInputStream` method would be good and handle the most general case, where we only have a byte stream. I prefer them to be all abstract as most browser "backends" have already optimized (or at least more robust) ways to deal with each type of input.

That said, right now `Browser` does not auto-detect charsets. The methods that don't receive a charset argument just call the charset variant with UTF-8 as default, a thing that could probably be improved later.